test/services/licences/supplementary/process-imported-licence.service.test.js

```diff
@@ -0,0 +1,274 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const Sinon = require('sinon')
+
+const { describe, it, beforeEach, afterEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const LicenceHelper = require('../../../support/helpers/licence.helper.js')
+
+// Things we need to stub
+const FetchExistingLicenceDetailsService = require('../../../../app/services/licences/supplementary/fetch-existing-licence-details.service.js')
+const PersistSupplementaryBillingFlagsService = require('../../../../app/services/licences/supplementary/persist-supplementary-billing-flags.service.js')
+
+// Thing under test
+const ProcessImportedLicenceService = require('../../../../app/services/licences/supplementary/process-imported-licence.service.js')
+
+describe('Process Imported Licence Service', () => {
+  let clock
+  let testDate
+  let PersistSupplementaryBillingFlagsServiceStub
+
+  beforeEach(() => {
+    testDate = new Date('2024-04-01')
+    clock = Sinon.useFakeTimers(testDate)
+    PersistSupplementaryBillingFlagsServiceStub = Sinon.stub(PersistSupplementaryBillingFlagsService, 'go').resolves()
+  })
+
+  afterEach(() => {
+    Sinon.restore()
+    clock.restore()
+  })
+
+  describe('when given a valid licenceId', () => {
+    let licence
+    let importedLicence
+
+    beforeEach(async () => {
+      licence = await LicenceHelper.add()
```

Comment:
If we are stubbing `FetchExistingLicenceDetailsService`, why do we need to create a real licence in the DB?